File: PAT/basic_level/1014_福尔摩斯的约会/get_date.py
from __future__ import print_function

def is_day(s):
	return s>='A' and s<='G'

def is_hour(s):
	return ( (s>='A' and s<='N') or s.isdigit() )

us_str, length, cnt = [0]*4, [0]*4, 0
day = ['MON', 'TUE', 'WED', 'THU', 'FRI', 'SAT', 'SUN']

for i in range(4):
	us_str[i] = raw_input()
	length[i] = len(us_str[i])

length[0] = min(length[0], length[1])
length[2] = min(length[2], length[3])

for i in range(length[0]):
	if is_day( us_str[0][i] ):
		if is_day( us_str[1][i] ):
			if us_str[0][i] == us_str[1][i]:
				print( day[ord(us_str[0][i])-ord('A')], '', end='' )
				cnt = i + 1
				break

for i in range(cnt, length[0]):
	if is_hour( us_str[0][i] ):
		if is_hour( us_str[1][i] ):
			if us_str[0][i] == us_str[1][i]:
				if us_str[0][i]>=0 and us_str[0][i]<=9:
					print( '0'+us_str[0][i], end='' )
				else:
					print( ord(us_str[0][i])-ord('A')+10, end='' )
				break

print(':', end='')

for i in range(length[2]):
	if us_str[2][i].isalpha():
		if us_str[2][i]==us_str[3][i]:
			if i<10:
				print('0', end='')
			print(i, end='')
			break


'''
3485djDkxh4hhGE 2984akDfkkkkggEdsb s&hgsfdk d&Hyscvnm

str.isalpha()
str.islower()
str.lower()
str.isupper()
str.upper()
str.isalnum()
str.isdigit()

'''

File: PAT/basic_level/1014_福尔摩斯的约会/get_date.cpp
#include <iostream>
using namespace std;

bool is_day(const char str){
	return str>='A' && str<='G';
}

bool is_hour(const char str){
	return ( (str>='A' && str<='N') || isdigit(str) );
}

int main(){
	string us_input[4];
	string day[]={"MON","TUE","WED","THU","FRI","SAT","SUN"};
	int len[4];
	int i=0, cnt=0;

	for( i=0; i<4; i++ ){
		cin>>us_input[i];
		len[i] = us_input[i].size();
	}

	if( len[0]>len[1] )
		len[0] = len[1];
	if( len[2]>len[3] )
		len[2] = len[3];

	for( i=0; i<len[0]; i++ )
		if( is_day(us_input[0][i]) )
			if( is_day(us_input[0][i]) )
				if( us_input[0][i]==us_input[1][i] ){
					cout<<day[ us_input[0][i]-'A' ]<<' ';
					cnt = i + 1;
					break;
				}

	for( i=cnt; i<len[0]; i++ )
		if( is_hour(us_input[0][i]) )
			if( is_hour(us_input[1][i]) )
				if( us_input[0][i]==us_input[1][i] )
					if( isdigit(us_input[0][i]) ){
						cout<<'0'<<us_input[0][i]<<':';
						break;
					}else{
						cout<<us_input[0][i]-'A'+10<<':';
						break;
					}

	for( i=0; i<len[2]; i++ )
		if( isalpha(us_input[2][i]) )
			if( us_input[2][i]==us_input[3][i] ){
				if( i<10 )
					cout<<'0'<<i;
				else
					cout<<i;
				break;
			}

	return 0;
}



/*
3485djDkxh4hhGE
2984akDfkkkkggEdsb
s&hgsfdk
d&Hyscvnm

*/
